File: app/src/main/java/com/siva/virtual/OpeningActivity.java
package com.siva.virtual;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatImageView;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import com.siva.virtual.util.Utils;

public class OpeningActivity extends AppCompatActivity {

    private Intent i;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        AppCompatImageView iv = findViewById(R.id.iv);

        SharedPreferences sp = getSharedPreferences(Utils.FIRST_LOGIN_DETAILS,MODE_PRIVATE);
        //checking whether user logging in for the first time or not
        if(sp.getBoolean(Utils.IS_FIRST_LOGIN,true)){
            i = new Intent(OpeningActivity.this,IntroActivity.class);
        }
        else
            i= new Intent(OpeningActivity.this,LoginScreen.class);

        Animation anim = AnimationUtils.loadAnimation(this,R.anim.scale_out);
        iv.startAnimation(anim);

        anim.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                startActivity(i);
                finish();
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
    }
}

File: app/src/main/java/com/siva/virtualWallet/OpeningActivity.kt
package com.siva.virtualWallet

import android.content.Intent
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatImageView
import com.siva.virtualWallet.util.Utils

class OpeningActivity : AppCompatActivity() {
    private var i: Intent? =
        null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        val iv =
            findViewById<AppCompatImageView>(R.id.iv)
        val sp =
            getSharedPreferences(Utils.FIRST_LOGIN_DETAILS, MODE_PRIVATE)
        //checking whether user logging in for the first time or not
        i =
            if (sp.getBoolean(Utils.IS_FIRST_LOGIN, true)) {
                Intent(this@OpeningActivity, IntroActivity::class.java)
            } else Intent(this@OpeningActivity, LoginScreen::class.java)
        val anim =
            AnimationUtils.loadAnimation(this, R.anim.scale_out)
        iv.startAnimation(anim)
        anim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                startActivity(i)
                finish()
            }

            override fun onAnimationRepeat(animation: Animation) {}
        })
    }
}